lib/runtime/src/pipeline/network/egress/push_router.rs

```diff
@@ -0,0 +1,182 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Arc<dyn AsyncEngine<SingleIn<AddressedRequest<In>>, ManyOut<Out>, Error>>;
+
+use async_trait::async_trait;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::{
+    marker::PhantomData,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+};
+
+use crate::{
+    component::{Client, Endpoint, EndpointSource},
+    engine::{AsyncEngine, Data},
+    pipeline::{AddressedPushRouter, AddressedRequest, Error, ManyOut, SingleIn},
+    traits::DistributedRuntimeProvider,
+};
+
+#[derive(Clone)]
+pub struct PushRouter<T, U>
+where
+    T: Data + Serialize,
+    U: Data + for<'de> Deserialize<'de>,
+{
+    // TODO: This shouldn't be pub, but lib/bindings/python/rust/lib.rs exposes it.
+    pub client: Client,
+    router_mode: RouterMode,
+    counter: Arc<AtomicU64>,
+    addressed: Arc<AddressedPushRouter>,
+    _phantom: PhantomData<(T, U)>,
```

Comment:
Can you document these fields, especially the `_phantom` field?

Reply:
Yes, thanks for keeping me honest! Definitely needed.